WT-6598 Add API to set number of hash buckets. (#5967) (v4.0 backport)

raviprakashgiri29 opened the pull request:
* WT-6598 Add API to set number of hash buckets.

* Add statistics so bucket sizes can be visible.

* Add checkpoint gather handle stats.

* Rename buckets to hash_size.

* Add test. Fix when we set the stats.

* Configure hash sizes even earlier.

* Minor changes from review comments.

* Run s_all.

* Require hash bucket sizes to be power of 2.

* Fix test to use power of 2.

* Fix value to be power of 2.

* Add test of invalid values.

* Move hash configuration processing earlier, before loading extensions.

* Typo.

(cherry picked from commit 06f0d05eebab678f9459fc15448df11a57b624cb)
(cherry picked from commit 3fc1d6a5245ae76530d79847d59ebefcbf39b419)

sueloverso commented:
Hmm, I see a call to `wt_free(conn->home)` in the `mongodb-4.0` branch already in `conn_handle.c:131`.  Are you somehow using an older version of the branch?

raviprakashgiri29 commented:
> Hmm, I see a call to `wt_free(conn->home)` in the `mongodb-4.0` branch already in `conn_handle.c:131`. Are you somehow using an older version of the branch?

Yes, It is part of mongodb-4.0 in conn_handle.c, but I deleted `__wt_free(session, conn->home);` by mistake while backporting the changes. Updated with the latest changes.

sueloverso reviewed (approved):
LGTM
